cmd/ipfs-cluster-service/main.go

```diff
@@ -643,14 +643,21 @@ func getStateManager() cmdutils.StateManager {
 }
 
 func assignRandomPorts(multiAddrs []*ma.Multiaddr) {
+	var prev int
 	for _, m := range multiAddrs {
 		components := []ma.Multiaddr{}
 		ma.ForEach(*m,
 			func(c ma.Component) bool {
 				code := c.Protocol().Code
 				var port int
 				if code == ma.P_TCP || code == ma.P_UDP {
-					ln, err := net.Listen("tcp", "0.0.0.0:")
+					var address string
+					if prev == ma.P_IP4 {
+						address = "0.0.0.0:"
```

Comment:
again, you need to listen on the actual interface specified by the previous component. You are hardcoding 0.0.0.0 where it could be 127.0.0.1 or something else...